staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -179,20 +179,41 @@ func (a *Authenticator) UpdateTransportConfig(c *transport.Config) error {
 		return &roundTripper{a, rt}
 	}
 
-	getCert := c.TLS.GetCert
-	c.TLS.GetCert = func() (*tls.Certificate, error) {
-		// If previous GetCert is present and returns a valid non-nil
-		// certificate, use that. Otherwise use cert from exec plugin.
-		if getCert != nil {
-			cert, err := getCert()
-			if err != nil {
-				return nil, err
+	if t, ok := c.Transport.(*http.Transport); ok && t != nil {
```

Comment:
what does this do when run with `--v=6`? might need a `func TransportFor(transport http.RoundTripper) (*http.Transport, error)` helper in pkg/util/net to unwrap nested debugging transports (see `DialerFor` as an example)

Reply:
Just tested, `--v=6` works fine.
But I agree with your point, someone can introduce a wrapper eventually, breaking this.
Added a `TransportFor` helper.